packages/autocomplete-js/src/render.tsx

```diff
@@ -90,90 +91,77 @@ export function renderPanel<TItem extends BaseItem>(
   dom.panel.classList.toggle('aa-Panel--touch', isTouch);
   dom.panel.classList.toggle('aa-Panel--stalled', state.status === 'stalled');
 
-  const sections = state.collections.map(({ source, items }, sourceIndex) => {
-    return createElement('section', {
-      key: sourceIndex,
-      className: classNames.source,
-      children: [
-        source.templates.header &&
-          createElement('div', {
-            className: classNames.sourceHeader,
-            children: [
-              source.templates.header({
-                createElement,
-                Fragment,
-                items,
-                source,
-                state,
-              }),
-            ],
-          }),
-        items.length === 0 && source.templates.empty
-          ? createElement('div', {
-              className: classNames.sourceEmpty,
-              children: [
-                source.templates.empty({
+  const sections = state.collections.map(({ source, items }, sourceIndex) => (
+    <section key={sourceIndex} className={classNames.source}>
+      {source.templates.header && (
+        <div className={classNames.sourceHeader}>
+          {source.templates.header({
+            createElement,
+            Fragment,
+            items,
+            source,
+            state,
+          })}
+        </div>
+      )}
+
+      {items.length === 0 && source.templates.empty ? (
+        <div className={classNames.sourceEmpty}>
+          {source.templates.empty({
+            createElement,
+            Fragment,
+            source,
+            state,
+          })}
+        </div>
+      ) : (
+        <ul
+          className={classNames.list}
+          {...propGetters.getListProps({
+            state,
+            props: autocomplete.getListProps({}),
+            ...autocompleteScopeApi,
+          })}
+        >
+          {items.map((item) => {
+            const itemProps = autocomplete.getItemProps({
+              item,
+              source,
+            });
+
+            return (
+              // @ts-ignore
+              // TypeScript assumes that events and other props are typed like
+              // React while it's typed as the DOM (like Preact). We ignore
+              // these errors.
+              //
+              // Example:
+              // Types of property 'onClick' are incompatible.
+              // Type '(event: MouseEvent) => void' is not assignable to type '(event: MouseEvent<HTMLLIElement, MouseEvent>) => void'.
```

Comment:
I couldn't find a way to tell TypeScript to not type-check with React. Let me know if you have an idea!

Reply:
It looks like there's a conflict when passing the function directly to the event but re-defining it with an arrow functions works.

Excuse the bad naming/explanation but here's what I mean: 

```jsx
const currentItemProps = propGetters.getItemProps({
    state,
    props: itemProps,
    ...autocompleteScopeApi,
});

return (
    <li
    key={itemProps.id}
    className={classNames.item}
    {...currentItemProps}
    onMouseDown={() => currentItemProps.onMouseDown}
    onMouseMove={() => currentItemProps.onMouseMove}
    onClick={() => currentItemProps.onClick}
    >
    {source.templates.item({
        createElement,
        Fragment,
        item,
        state,
    })}
    </li>
);
```

Reply:
This TS config fixed it: c90c7794e7a204bd3c033a4339712a8be486535e.